File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    p = progresses
    s = speeds
    # 작업이 모두 사라질 때까지 반복한다.
    while p :
        cnt = 0
        # 100 이상이 된 작업이 있으면 cnt += 1을 해주고 작업 목록에서 뺀다.
        while p and p[0] >= 100:
            cnt += 1
            p.pop(0)
            s.pop(0)
        # 하루마다 진행도를 올린다.
        for i in range(len(p)):
            p[i] += s[i]
        # cnt가 1 이상이면 답 목록에 추가하고 다음 날로 넘어간다.
        if cnt:
            answer.append(cnt)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];

    while(speeds.length > 0) {
        for(let i in speeds) {
            if(progresses[i] < 100) {
                progresses[i] += speeds[i];
            }
        }

        let deploy_count = 0;
        while(progresses[0] >= 100) {
            progresses.shift();
            speeds.shift();
            deploy_count++;
        }
        if(deploy_count > 0) {
            answer.push(deploy_count);
        }
    }

    return answer;
}